core/deployment/src/main/java/io/quarkus/deployment/QuarkusAugmentor.java

```diff
@@ -74,6 +76,12 @@
     }
 
     public BuildResult run() throws Exception {
+        Pattern pattern = Pattern.compile("(?:1\\.)?(\\d+)(?:\\..*)?");
+        Matcher matcher = pattern.matcher(System.getProperty("java.version", ""));
+        if (matcher.matches() && Integer.parseInt(matcher.group(1)) < 11) {
+            log.warn("Using Java versions older than 11 to build"
+                    + " Quarkus applications is deprecated and will be disallowed in a future release!");
```

Comment:
I usually refrain from using exclamation marks in user messages but feel free to ignore.

Reply:
TBH I was wondering if it should be bigger... like, a big banner

```
***********************************************************************
*** WARNING WARNING WARNING WARNING WARNING WARNING
*** 🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥
***********************************************************************

```

Reply:
with animated, red flames :)

Reply:
```
/ Using Java versions older than 11 to   \
| build Quarkus applications is          |
| deprecated and will be disallowed in a |
\ future release. I mean, moo            /
 ----------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||
```

Reply:
Ok, well, I will merge the version with the exclamation mark before someone comes with more brilliant ideas :).